setup.py

```diff
@@ -297,6 +298,10 @@ def get_base_commit_in_main_branch(self) -> str:
             ]).decode("utf-8")
             upstream_main_commit = json.loads(resp_json)["sha"]
 
+            # In Docker build context, .git may be immutable or missing.
+            if envs.VLLM_DOCKER_BUILD_CONTEXT:
+                return upstream_main_commit
+
```

Comment:
This will be problematic as the main commit might not have the wheels ready (e.g. when it is just merged).

Reply:
I guess the same problem will be if the PR merge base is not compatible with latest main/nightly. we can address this as a follow up.